scripts/morty/plotter.py

```diff
@@ -0,0 +1,204 @@
+        if not os.path.isdir(self.imdir):
+            os.mkdir(self.imdir)
+        return
+    
+    def parasitic_plot(self, parasitic_data):
+        """
+        Plots the parasitic absorption density
+
+        Parameters
+        ----------
+        parasitic_data : dict
+            key : str
+                Name of variable
+        
+        """
+        parasitic_use_data = parasitic_data['parasitic']
+        for nuclide_index in range(np.shape(parasitic_use_data)[1]):
+            for method_i in range(len(parasitic_data['parasitic'])):
+                method_name = parasitic_data['labs'][method_i]
+                x = self.run_params['times']
+                y = parasitic_use_data[method_i][nuclide_index]
+                lab = self.data_params['tracked_nucs'][nuclide_index]
+                plt.plot(x, y, label=f'{method_name} {lab} Captures')
+            plt.yscale(self.yscale)
+            plt.xlabel('Time [s]')
+            plt.ylabel('Concentration [atoms/cc]')
+            plt.legend()
+            plt.tight_layout()
+            plt.savefig(f'{self.imdir}parasitic_absorption_{nuclide_index}.png')
+            plt.close()
+        return
+    
+    def plot_time(self, data_dict, spatial_eval_node=None):
+        """
+        Plots average data over time where `data_dict` contains all
+         necessary information
+
+        Parameters
+        ----------
+        data_dict : dict
+            key : str
+                Variable name
+        spatial_eval_node : int (optional)
+            Spatial node index to evaluate    
+        """
+        num_nucs = []
+        for i, x in enumerate(data_dict['xs']):
+            num_nucs.append(np.shape(data_dict['ys'][i])[2])
+        num_nucs = max(num_nucs)
+        for nuclide_i in range(num_nucs):
+            for i, x in enumerate(data_dict['xs']):
+                if type(spatial_eval_node) == type(None):
+                    try:
+                        y = data_dict[f'spat_avg_y_method{i}_nuc{nuclide_i}']
+                    except KeyError:
+                        continue
+                    ending = 'avg'
+                else:
+                    try:
+                        y = data_dict['ys'][i][:, spatial_eval_node, nuclide_i]
+                    except IndexError:
+                        continue
+                    ending = f'spat{spatial_eval_node}'
+                lab = data_dict[f'lab_method{i}_nuc{nuclide_i}']
+                plt.plot(x, y, label=lab)
+            plt.xlabel(data_dict['xlab'])
+            plt.ylabel(data_dict['ylab'])
+            plt.yscale(self.yscale)
+            plt.legend()
+            try:
+                plot_main_name = f'{self.imdir}{data_dict['savename']}'
+                plt.savefig(f'{plot_main_name}_{nuclide_i}_{ending}.png')
+            except UnboundLocalError:
+                pos = spatial_eval_node
+                print(f'No data for {nuclide_i} figure at spatial index {pos}')
```

Comment:
Just something to consider: I'm a big fan of logger and Warnings for this kind of thing.